File: examples/http.java
import org.slf4j.Logger;
import com.fizzed.blaze.Contexts;
import com.fizzed.blaze.util.MutableUri;
import java.net.URI;
import static com.fizzed.blaze.Https.*;

public class http {
    static final private Logger log = Contexts.logger();

    public void main() throws Exception {
        URI uri = MutableUri.of("http://jsonplaceholder.typicode.com/comments")
            .query("postId", 1)
            .toURI();
        
        String output = httpGet(uri.toString())
            .addHeader("Accept", "application/json")
            .runCaptureOutput()
            .toString();
        
        log.info("Quote of the day JSON is {}", output);
    }
    
}

File: examples/http.js
/* global Packages, Contexts, MutableUri, Request */

// nashorn recommended method of importing classes
var Imports = new JavaImporter(
    Packages.org.slf4j.Logger,
    Packages.com.fizzed.blaze.Contexts,
    Packages.com.fizzed.blaze.util.MutableUri,
    Packages.com.fizzed.blaze.Https);
    
with (Imports) {
    
    var log = Contexts.logger();

    var main = function() {
        var uri = MutableUri.of("http://jsonplaceholder.typicode.com/comments")
            .query("postId", 1)
            .toURI();

        var output = Https.httpGet(uri.toString())
            .addHeader("Accept", "application/json")
            .runCaptureOutput()
            .toString();

        log.info("Quote of the day JSON is {}", output);
    };
    
}